### ISSUE TITLE
User login problem

### ISSUE THREAD

**Issue by BoikoArtem30091994** [TheM4hd1/SwiftyInsta#200]
> For future reference… the new behaviour is this.
> (@mkeshnoda)
> 
> ```swift
> class LoginViewController: UIViewController {
>     var webView: InstagramLoginWebView? {
>         didSet {
>             oldValue?.removeFromSuperview()
>             guard let webView = webView else { return }
>             // add to view hierarchy (NECESSARY).
>             view.addSubview(webView)
>             // always request log in page when setting a new web view.
>             webView.loadInstagramLogin()
>         }
>     }
>     
>     override func viewDidLoad() {
>         super.viewDidLoad()
>         // create a new `InstagramLoginWebView`.
>         InstagramLoginWebView.create(with: view.bounds,
>                                      completionHandler: { _, _ in /* deal with login and failures */ },
>                                      returnHandler: { [weak self] in self?.webView = $0 })
>     }
> }
> ```

I did exactly the same but after logging-in **didSuccessfullyLogIn** and **completionHandler** is not working 😭 

Just did a little debugging and found out that i can't login because of  `filtered.count  = 1`. `isUserLoggedIn(instagramCookies: [HTTPCookie]?)` recieves only 2 cookies
```
let filtered = cookies.filter {
            !$0.value.isEmpty
                && $0.properties?[.init(rawValue: "Created")] as? TimeInterval ?? 0 > startedAt.timeIntervalSinceReferenceDate
                && ($0.name == "ds_user_id" || $0.name == "csrftoken" || $0.name == "sessionid") }
        guard filtered.count >= 3 else { return }
```

_Originally posted by @mkeshnoda in https://github.com/TheM4hd1/SwiftyInsta/issues/75#issuecomment-519862211_

**Comment by BoikoArtem30091994** [comment]
Hello. Faced the same problem. The amount of cookies returned is different. Any solution for this?

**Comment by sbertix** [comment]
It used to be an issue related to the user agent, and I'm assuming it might still be the case.\
Do you mind checking with [**Swiftagram**](https://github.com/sbertix/Swiftagram) if it happens too (just run the **Followers** example, no need to set it up yourself), so I can understand whether it's about Instagram spam accounts detection or just implementation?

**Comment by BoikoArtem30091994** [comment]
Ok, I'll try to run this.

**Comment by BoikoArtem30091994** [comment]
@sbertix Managed to run followers and it seems to log in successfully. What follows from this?

**Comment by sbertix** [comment]
Sorry I didn't manage to get back at you earlier @BoikoArtem30091994\
It means it's probably an issue with **SwiftyInsta**, as it's not found in another implementation (e.g. **Swiftagram**).\
I'm assuming the bug fix was reverted in #183.

There's a PR still open by @bariscck #190, fixing changes he made in #183, so I'd let him fix it, if he wants to.\
Unfortunately, nowadays, while actively working on **Swiftagram**, I'm just focusing on **critical bugs** (i.e. they need to at least be replicable) for **SwiftyInsta**, and it does not seem to be the case here. Sorry.

**Comment by sbertix** [comment]
I'm closing this and marking it as `wontfix` for now. If somebody decides to work on it they can simply reopen it.
